code/bmpman/bmpman.cpp

```diff
@@ -2576,6 +2609,61 @@ void bm_page_in_start() {
 	gr_bm_page_in_start();
 }
 
+/**
+ * Stuffs Pathtypes with an additional path leading to a subdir of the same name as the given file (minus file extension).
+ * Used for allowing frames of .eff animations to be found in their own subdirectories.
+ *
+ * Example: if called with "debris.eff" and that file is in data/maps, then Pathtypes[CF_TYPE_TEMP_SUBDIR_LOOKUP].path
+ * will be stuffed with data/maps/debris, allowing data/maps/debris/debris_0000.dds (etc) to be found
+ */
+bool set_temp_subdir_pathtype(const char *filename) {
+	// Gets the absolute path of the file, removes the part that is identical with the
+	// absolute root path and appends the filename to it (without extension), giving us
+	// a relative path that can be temporarily stuffed into Pathtypes so the frames can be found
+
+	char fullpath[MAX_PATH]; // Absolute path to the given file
+	char fullrootpath[MAX_PATH]; // Absolute root path
+	size_t size, offset; // Needed for cf_find_file_location, but not used
+
+	cf_create_default_path_string(fullrootpath, sizeof(fullrootpath) - 1, CF_TYPE_ROOT, NULL);
+	if (!cf_find_file_location(filename, CF_TYPE_ANY, sizeof(fullpath)-1, fullpath, &size, &offset)) {
+		mprintf(("BMPMAN: Failed to set temporary pathtype for %s: file not found!\n", filename));
+		return false;
+	}
+
+#ifndef NDEBUG
+	if (strlen(fullpath) < strlen(fullrootpath)) {
+		mprintf(("BMPMAN: Path to %s was shorter than the root path, programmer thought it could never happen!\n%s\n%s\n", filename, fullpath, fullrootpath));
```

Comment:
If this should never happen then an `Assertion` might be the better choice here.